Fix dev branch (#248)
LiskHQ/lisk-docs: examples.md

Welcome to the step-by-step guide of creating the Hello World application with Lisk Alpha SDK.

> A simple App, showcasing a minimal setup of a blockchain application with 1 [custom transaction](custom-transactions.md) type: the "Hello" transaction.

Reviewer: ```suggestion
A simple App, showcasing a minimal setup of a blockchain application with 1 [custom transaction](custom-transactions.md) type: the "HelloWorld" transaction.
```

> The purpose of Hello World application is to explain how to use and how to implement custom transactions with the Lisk SDK. 

Reviewer: ```suggestion
The purpose of Hello World application is to show how to implement custom transactions with the Lisk SDK. 
```

> The implementation is saving the string value of the "hello" transaction's asset property to the asset property of the sender's account.

Reviewer: ```suggestion
The implementation is saving the string value of the "helloWorld" transaction's asset property to the asset property of the sender's account.
```

## Cashback App

A simple application which rewards its users for sending tokens. 

> Check out the full code example for the [Cashback App on Github](https://github.com/LiskHQ/lisk-sdk-test-app).

### 1. Set up Lisk SDK

First, you need to set up the Lisk SDK. Please follow the instructions in the [Lisk SDK - Usage](../lisk-sdk/introduction.md#usage) section.

### 2. Configure the application

Next, let's configure the application, to provide basic information about the app we are going to build:

```js
//index.js
const { Application, genesisBlockDevnet, configDevnet } = require('lisk-sdk'); // require application class, the default genesis block and the default config for the application

const app = new Application(genesisBlockDevnet, configDevnet); // create the application instance
```

> In the `line 2`, we require the needed dependencies from the `lisk-sdk` package.

Reviewer: ```suggestion
On `line 2`, we require the needed dependencies from the `lisk-sdk` package.
```

The most important one is the `Application` class, which is used in `line 4` to create the application instance.
The application instance will start the whole application at the bottom of `index.js`.

In `line 4` , the application instance gets initialized.
By passing the parameters for the [genesis block](../lisk-sdk/configuration.md) and the [configuration template](https://github.com/LiskHQ/lisk-sdk/blob/development/sdk/src/samples/config_devnet.json), the application is configured with most basic configurations to start the node.

### 3. Create a new transaction type

Now, we want to create a new [custom transaction type](custom-transactions.md) `CashbackTransaction`: 
It extends the pre-existing transaction type `TransferTransaction`.
In difference to the normal `TransferTransaction`, the Cashback transaction type pays out a 10% bonus reward to the sender of any Cashback transaction type.

So e.g. if Alice sends 100 token to Bob as a Cashback transaction, Bob would receive the 100 token and Alice would receive additional 10 tokens as a cashback.

> > If you compare the methods below with the methods we implemented for the `HelloTransaction`, you will notice, that we implement less methods for the `CashbackTransaction`.

Reviewer: ```suggestion
> If you compare the methods below with the methods we implemented for the `HelloWorldTransaction`, you will notice, that we implement less methods for the `CashbackTransaction`.
```
If we want to move to the HelloWorld name

> This is because we extend the `CashbackTransaction` from an already existing transaction type `TransferTransaction`.
> As a result, all required methods are implemented already inside the `Transfertransaction` class, and we only need to overwrite/extend explicitely the methods we want to customize.

Now, let's create a new file `cashback_transaction.js`, which is defining the new transaction type `CashbackTransaction`:

```js
//cashback_transaction.js
const {	TransferTransaction, BigNum } = require('lisk-sdk'); // import the TransferTransaction class and Bignum from the Lisk SDK package

class CashbackTransaction extends TransferTransaction { // let the CashbackTransaction become a child class of TransferTransaction

// add the all methods described below here

}

module.exports = CashbackTransaction;
``` 

> You find a version of [the complete `cashback_transaction.js` file](https://github.com/LiskHQ/lisk-sdk-examples/blob/development/cashback/transactions/cashback_transaction.js) on Github.

#### TYPE

> Set the Cashback transaction TYPE to `11`.

Reviewer: Again should reserve more than 10 for Lisk?

The first 10 types, from `0-9` is reserved for the default Lisk Network functions.
Type `10` was used previously for the `HelloTransaction`, so we set it to `11`, but any other integer value (that is not already used by another transaction type) is a valid value.
```js
static get TYPE () {
    return 11;
}
```

#### applyAsset

> It sends the transaction amount to the recipient and substracts 1/10 of the transaction amount from the senders account balance

Reviewer: ```suggestion
The CashbackTransaction adds an inflationary 10% to senders account.
```

> Invoked as part of the `apply` step of the BaseTransaction and block processing.  

Reviewer: ```suggestion
```
I think this sentence is a little confusing

Reviewer 2: There is an apply step on base transaction as well, applyAsset is being called as part of it.

    };
    store.account.set(sender.address, updatedSender); // push updated account back to database

    return [];
}
```

### 4. Register the new transaction type

Right now, your project should have the following file structure:

```bash
/cashback-app # root directory of the application
/cashback-app/cashback_transaction.js # the custom transaction, created in step 3
/cashback-app/index.js # the index file of your application, created in step 1, extended in step 2 and 4
/cashback-app/node_modules/ # project dependencies, created in step 1
/cashback-app/package.json # project manifest file, created in step 1
```

Add the new transaction type to your application, by registering it to the application instance:

```js
//index.js
const { Application, genesisBlockDevnet, configDevnet} = require('lisk-sdk'); // require application class, the default genesis block and the default config for the application
const CashbackTransaction = require('./cashback_transaction'); // require the newly created transaction type 'CashbackTransaction'

const app = new Application(genesisBlockDevnet, configDevnet); // create the application instance

> app.registerTransaction(11, CashbackTransaction); // register the 'CashbackTransaction' 

Reviewer: ```suggestion
app.registerTransaction(CashbackTransaction); // register the 'CashbackTransaction' 
```
According @nazarhussain we can remove the number, is this correct @MaciejBaj ?
